library/pk.c

```diff
@@ -347,22 +347,88 @@ int mbedtls_pk_verify_ext( mbedtls_pk_type_t type, const void *options,
     if( ! mbedtls_pk_can_do( ctx, type ) )
         return( MBEDTLS_ERR_PK_TYPE_MISMATCH );
 
-    if( type == MBEDTLS_PK_RSASSA_PSS )
+    if( type != MBEDTLS_PK_RSASSA_PSS )
     {
+        /* General case: no options */
+        if( options != NULL )
+            return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+        return( mbedtls_pk_verify( ctx, md_alg, hash, hash_len, sig, sig_len ) );
+    }
+
 #if defined(MBEDTLS_RSA_C) && defined(MBEDTLS_PKCS1_V21)
-        int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-        const mbedtls_pk_rsassa_pss_options *pss_opts;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
```

Comment:
Minor: I remember from my review comments that variables should be declared at the beginning of the function or at least at the beginning of the block.
Not sure if this is very important.

Reply:
That was only the case for the 2.16 branch (and therefore for bugfixes and tests that had to be backported there), which had to work with C89 compilers. Now that 2.16 is discontinued, we can use many C99 features, including variable declarations after statements and in for loops.